charms/reactive/bus.py

```diff
@@ -444,19 +444,25 @@ def _load_module(root, filepath):
     return importlib.import_module(package + module)
 
 
-def _register_handlers_from_file(root, filepath):
-    exec_whitelist = ('', '.py', '.sh')
-    no_exec_blacklist = (
-        'makefile', '.gitignore',
-        'copyright', 'license')
+def _is_external_handler(filepath):
+    if not os.access(filepath, os.X_OK):
+        return False
     _, ext = os.path.splitext(filepath)
-    if ext not in exec_whitelist:
-        # Don't load handlers unless they match the whitelist
-        return
-    if filepath.lower().endswith(no_exec_blacklist):
-        # Don't load handlers which match the blacklist
-        return
+    if ext not in ('', '.sh'):
```

Comment:
Definitely an edge case, but do we need to worry about `*.bash`, `*.fish`, `*.zsh`, etc?

Reply:
Given my track record so far, probably.  ;)  But my preference is to be too restrictive by default because, as I said before, I consider this a deprecated and mostly unused feature anyway.